Initial Gazebo Implementation of 2024 Robot URDF

Isopod00 opened the pull request:
Ready for review!

jmblixt3 committed:
Update robot base frame and RViz configuration
Setup work to creat robot visualization

umnrobotics committed:
Initial testing of Gazebo world

umnrobotics committed:
Added more rocks :)

Isopod00 committed:
Added a basic robot

Isopod00 committed:
fixed base_link

Isopod00 committed:
Updated the README

Isopod00 committed:
Added the ros_gz submodule

Isopod00 committed:
updated README

jmblixt3 committed:
Remove ros_gz source use binary

jmblixt3 committed:
Add submodule for ros_sim

jmblixt3 committed:
Add submodule for gazebo

Isopod00 committed:
Updated the size of our URDF

Isopod00 committed:
Merge pull request #132 from GOFIRST-Robotics/main

Sync with the main branch

umnrobotics committed:
Merge remote-tracking branch 'origin/main' into gazebo-testing

umnrobotics committed:
update isaaac ros common

Isopod00 committed:
Merge remote-tracking branch 'origin/main' into gazebo-testing

Isopod00 committed:
updated the submodule

Isopod00 committed:
Merge branch 'main' into gazebo-testing

pwmarshall committed:
swapped x and y speed

Isopod00 committed:
Merge branch 'main' into gazebo-testing

pwmarshall committed:
gazebo fun

pwmarshall committed:
Merge branch 'gazebo-testing' of https://github.com/GOFIRST-Robotics/Lunabotics-2024 into gazebo-testing

pwmarshall committed:
Better URDF Robot

pwmarshall committed:
basic movement

Isopod00 committed:
Merge branch 'main' into gazebo-testing

pwmarshall committed:
drivetrain gazebo publishing

pwmarshall committed:
robot works?

pwmarshall committed:
simulation variable

Isopod00 committed:
Undo changes from the URDF branch

Isopod00 committed:
whitespace changes

Isopod00 committed:
another whitespace change

pwmarshall committed:
simulation ros parameter

pwmarshall committed:
Merge branch 'gazebo-testing' of https://github.com/GOFIRST-Robotics/Lunabotics-2024 into gazebo-testing

pwmarshall committed:
useless comment deletion

Isopod00 committed:
Moved gazebo stuff out of the umn dockerfile

Isopod00 committed:
Update README.md

Isopod00 committed:
update isaac_ros_common ref

pwmarshall committed:
Clean up :thumbs-up:

pwmarshall committed:
Merge branch 'gazebo-testing' of https://github.com/GOFIRST-Robotics/Lunabotics-2024 into gazebo-testing

Isopod00 commented on src/drivetrain/drivetrain/drivetrain_node.py:
```diff
@@ -11,22 +11,28 @@
 from rovr_interfaces.srv import Stop, Drive, MotorCommandSet
 from rovr_interfaces.msg import AbsoluteEncoders
 
+simulation = True
```
we don't need this variable here anymore right?

pwmarshall replied:
yeah i have a commit removing it

pwmarshall committed:
useless stl

pwmarshall commented on .gitmodules:
wait but no more submodule?

Isopod00 replied:
idk, are we keeping the ros_gz_moon submodule or not @jmblixt3? It's probably fine either way tbh

jmblixt3 replied:
Unless we move peters code over

Isopod00 replied:
The ros_gz_moon submodule has been removed and Peter's code was moved over

Isopod00 committed:
Merge branch 'main' into gazebo-testing

pwmarshall committed:
remove unused variable

pwmarshall committed:
Merge branch 'gazebo-testing' of https://github.com/GOFIRST-Robotics/Lunabotics-2024 into gazebo-testing

pwmarshall commented on config/drivetrain_config.yaml:
Wasn't sure if you wanted this to be True or false

Isopod00 replied:
Good point, can you please make it false by default

pwmarshall committed:
remove unneeded stl files

pwmarshall committed:
simulation false when launched with launch file

Isopod00 committed:
Delete ros_gz_moon from .gitmodules

Isopod00 committed:
Merge branch 'main' into gazebo-testing

pwmarshall commented:
to test on ubuntu (and generally linux)
with ros installed:
`sudo apt-get install ros-humble-ros-gz` to get gazebo
`sudo apt-get install ros-humble-teleop-twist-keyboard` to get keyboard control for ros

then run from local repo
```
colcon build --symlink-install --packages-up-to ros_gz_launch
source install/setup.bash
ros2 launch ros_gz_launch master_ASM.launch.py
```
```
colcon build --symlink-install --packages-up-to drivetrain
source install/setup.bash
ros2 run drivetrain drivetrain_node
```
and 
```
source /opt/ros/humble/setup.bash
ros2 run teleop_twist_keyboard teleop_twist_keyboard
```

These will launch Gazebo, DrivetrainNode, and keyboard controls in that order.

pwmarshall reviewed (approved):
Made it and tested, is good

Isopod00 commented on README.md:
````diff
@@ -82,6 +87,11 @@ To normalize line endings in git, use the command:
 ```
 
+## Gazebo Installation
+Install Gazebo Harmonic [here](https://staging.gazebosim.org/docs/harmonic/install_ubuntu).
````
Change this since we are using Gazebo Fortress, not Harmonic

pwmarshall committed:
Update readme to have correct verion of gazebo

pwmarshall committed:
Merge branch 'gazebo-testing' of https://github.com/GOFIRST-Robotics/Lunabotics-2024 into gazebo-testing

Isopod00 committed:
Updated gazebo launch file

jmblixt3 committed:
Update README.md

Edit readme to explain how to run commands, instructions on installation were removed since container was tested

Isopod00 committed:
README Update

Isopod00 reviewed (commented):
Looks good to me
